python/tests/test_highlevel.py

```diff
@@ -1834,20 +1835,22 @@ def verify_tables_api_equality(self, ts):
     def test_simplify(self):
         num_mutations = 0
         for ts in get_example_tree_sequences():
-            self.verify_tables_api_equality(ts)
-            self.verify_simplify_provenance(ts)
-            n = ts.num_samples
-            num_mutations += ts.num_mutations
-            sample_sizes = {0}
-            if n > 1:
-                sample_sizes |= {1}
-            if n > 2:
-                sample_sizes |= {2, max(2, n // 2), n - 1}
-            for k in sample_sizes:
-                subset = random.sample(list(ts.samples()), k)
-                self.verify_simplify_topology(ts, subset)
-                self.verify_simplify_equality(ts, subset)
-                self.verify_simplify_variants(ts, subset)
+            # Can't simplify edges with metadata
+            if ts.tables.edges.metadata_schema == tskit.MetadataSchema(schema=None):
```

Comment:
This isn't a great pattern I think - I've opened #2079 to track. We shouldn't have to create a new metadata schema object to determine whether the current one is set or not.